test/jdk/javax/print/CUPSPrinterImageableAreaTest.java

```diff
@@ -0,0 +1,221 @@
+
+    private static PrintService printService;
+    private static final List<MediaSizeName> ALLOWED_MEDIA_LIST = List.of(MediaSizeName.ISO_A4, MediaSizeName.NA_LETTER);
+    private static final double DPI = 72.0;
+    private static final double MM_PER_INCH = 2.54;
+    private static final String INSTRUCTIONS_FORMAT = """
+            <html>
+                <h3>The <b>'%s'</b> paper size should be used for this test.</h3>
+                <div>
+                The test checks that the media margins fetched from the printer's PPD file are correct.<br>
+                Press the '<b>Print sample</b>' button to print a test page.<br>
+                A passing test will print the page with a black rectangle along the printable area.
+                Ensure that all sides of the rectangle are printed.<br>
+                Expected margins:
+                <ul>
+                    <li>left: %.1f mm.</li>
+                    <li>bottom: %.1f mm.</li>
+                    <li>right: %.1f mm.</li>
+                    <li>top: %.1f mm.</li>
+                </ul>
+                Click '<b>Pass</b>' button, or click '<b>Fail</b>' button if the test failed.
+                </div>
+            <html>
+            """;
+
+
+    public static void main(String[] args) throws Exception {
+        printService = PrintServiceLookup.lookupDefaultPrintService();
+        if (printService == null) {
+            throw new RuntimeException("Print service not found");
+        }
+        MediaSizeName testMedia = getTestMediaSizeName();
+        PageFormat pageFormat = createTestPageFormat(testMedia);
+
+        // margins: left, bottom, right, top
+        double[] margins = new double[] {
+                pageFormat.getImageableX(),
+                pageFormat.getHeight() - pageFormat.getImageableHeight() - pageFormat.getImageableY(),
+                pageFormat.getWidth() - pageFormat.getImageableWidth() - pageFormat.getImageableX(),
+                pageFormat.getImageableY()
+        };
+
+        final String instructions = INSTRUCTIONS_FORMAT.formatted(testMedia.toString(),
+                inchesToMM(margins[0]), inchesToMM(margins[1]), inchesToMM(margins[2]), inchesToMM(margins[3]));
+
+        PassFailJFrame.builder()
+                .instructions(instructions)
+                .testUI(createTestUI(pageFormat))
+                .columns(55)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private static JFrame createTestUI(PageFormat pageFormat) {
+        final JFrame f = new JFrame("CUPS Printer imageable area test");
+        JButton printButton = new JButton("Print sample");
+        printButton.setPreferredSize(new Dimension(200, 80));
+        printButton.addActionListener((e) -> {
+            printButton.setEnabled(false);
+            PrinterJob job = PrinterJob.getPrinterJob();
+            job.setPrintable(new RectPrintable(), pageFormat);
+            try {
+                job.print();
```

Comment:
it now goes to default printer..maybe also give an option to show a printdialog and let user choose the printer..I guess choosing PDF-printer will not show the issue?